syft/pkg/cataloger/cpe.go

```diff
@@ -323,15 +343,152 @@ func groupIDFromPomProperties(p pkg.Package) string {
 		return ""
 	}
 
-	return metadata.PomProperties.GroupID
+	artifactID := strings.TrimSpace(metadata.PomProperties.ArtifactID)
+	if startsWithDomain(artifactID) && len(strings.Split(artifactID, ".")) > 1 {
+		// there is a strong indication that the artifact ID is really a group ID, don't use it
+		return ""
+	}
+	return artifactID
 }
 
-func shouldConsiderGroupID(groupID string) bool {
-	if groupID == "" {
-		return false
+func groupIDsFromJavaPackage(p pkg.Package) (groupIDs []string) {
```

Comment:
What is the purpose of naming the function's result parameter?

Reply:
(ditto this question for other occurrences below, but I'll keep to just one PR comment for all of them)

Reply:
this was for convenience for simpler functions, no other reason

Reply:
Interesting, okay thanks! 👍